api/graphql/auth/auth.go

```diff
@@ -72,31 +88,43 @@ func UserFromContext(ctx context.Context) *models.User {
 	return raw
 }
 
-func AuthWebsocketInit(db *gorm.DB) func(context.Context, transport.InitPayload) (context.Context, error) {
-	return func(ctx context.Context, initPayload transport.InitPayload) (context.Context, error) {
+func AuthWebsocketInit(db *gorm.DB) func(context.Context, transport.InitPayload) (context.Context, *transport.InitPayload, error) {
+	return func(ctx context.Context, initPayload transport.InitPayload) (context.Context, *transport.InitPayload, error) {
 
 		bearer, exists := initPayload["Authorization"].(string)
 		if !exists {
-			return ctx, nil
+			return ctx, nil, nil
 		}
 
 		token, err := TokenFromBearer(&bearer)
 		if err != nil {
-			log.Printf("Invalid bearer format (websocket): %s\n", bearer)
-			return nil, err
+			log.Error(ctx, "Invalid bearer format (websocket)", "error", err)
+			return nil, nil, err
+		}
+
+		loaders := dataloader.For(ctx)
+		if loaders == nil {
+			loaders = dataloader.NewLoaders(db)
+			ctx = context.WithValue(ctx, dataloader.GetLoadersKey(), loaders)
 		}
```

Comment:
I suggest returning an error here if dataloader doesn't exist. If dataloader doesn't exist, it means `dataloader.Middleware()` doesn't run and there is no loader injected. Blindly creating a new one just repeats the logic of `dataloader.Middleware()`. Since all middleware is constructed by the code, it should depend on `dataloader.Middleware()` but not repeat it again.

And please remove `dataloader.NewLoader()`. All loader should be constructed by `dataloader.Middleware()`.